src/compiler/sys.ts

```diff
@@ -1662,6 +1662,12 @@ namespace ts {
             }
 
             function fileSystemEntryExists(path: string, entryKind: FileSystemEntryKind): boolean {
+                // fs.statSync can be slow if path doesn't exist since Node.js generates a stack trace. Adding an
+                // fs.existsSync guard here can improve performance by reducing the number of times fs.statSync throws.
+                if (!_fs.existsSync(path)) {
+                    return false;
+                }
+
                 try {

```

Comment:
Another option is to just return using existsSync and ignore directory or file and make our code more tolerant (it might already be) for finding file instead of directory or vice versa. That could give us better perf in both cases.

Reply:
Definitely. I was also wondering how necessary it was to check for file versus folder existence.

I'm guessing the current behavior of this function is this way so folders ending with `.ts` get ignored. Is that an edge case where a behavior change would be acceptable?

Reply:
If you make a change we can see what user tests break and see what happens

Reply:
Ended up not doing this and favoring `Error.stackTraceLimit = 0` instead. I assume we wanted to try using `existsSync` entirely since we were worried about perf in the path exists case. Setting `Error.stackTraceLimit = 0` should improve performance when files don't exist with no change to checking files that do exist.

Reply:
Nice!